### ISSUE TITLE
Add Ruby (Rails) rule for unsafe direct object reference

### ISSUE THREAD

**Issue by elsapet** [Bearer/bearer-rules#55]
If model A belongs to model B, it is advised against calling `find` or `find_by` methods directly on model A itself. For example, if a User belongs to an Organization, the following is an unscoped find as it could lead to us retrieving a user that does not belong to the current organization or context

```ruby
User.find(user_id)
```

and instead we should scope this to the organization

```ruby
current_organization.users.find(user_id)
```

We should add a rule to guard against this security risk, but we should also avoid false positives (e.g. matching on any instance of a direct `find` or `find_by` method call)

See 
- [Brakeman Unscoped Find Warning](https://brakemanscanner.org/docs/warning_types/unscoped_find/)
- [OWASP insecure direct object reference prevention cheat sheet](https://cheatsheetseries.owasp.org/cheatsheets/Insecure_Direct_Object_Reference_Prevention_Cheat_Sheet.html)
